### PR TITLE
UIDATIMP-1333: Improve mapping profile for Orders

### PR DESCRIPTION
**Pull request by mariia-aloshyna**
## Purpose

To adjust the mapping profile for Orders on UI-side.

## Approach

- All changes with fields are described in UIDATIMP-1333
- Implemented `useOrganizationValue` hook to display the correct org value on the view screen

## Refs

https://issues.folio.org/browse/UIDATIMP-1333

### CODE REVIEW

**Review by VRohach — APPROVED** [review]

File: src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/CostDetails.js [hunk 1]
```diff
@@ -106,16 +107,19 @@ export const CostDetails = ({
...
-              setIsSetExchangeDisabled(!(e.target.value === BOOLEAN_ACTIONS.ALL_FALSE));
+            onChange={() => {
+              setIsUseExchangeChecked(prevState => !prevState);
+              setIsSetExchangeDisabled(prevState => {
+                if (!prevState) {
+                  setReferenceTables(getFieldName(51), '');
+                }
+
+                return !prevState;
+              });
```
Comment (artem-blazhko) [inline src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/CostDetails.js:120]: I would create a separate function outside `return` block to get rid of possible unnecessary rerenders.

File: src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/EResourcesDetails.js [hunk 2]
```diff
@@ -88,7 +88,7 @@ export const EResourcesDetails = ({
               <FieldOrganization
                 id={accessProviderId}
                 setReferenceTables={setReferenceTables}
-                name={getFieldName(68)}
+                name={getFieldName(64)}
```
Comment (artem-blazhko) [inline src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/EResourcesDetails.js:91]: Would be great to have some constants for those numbers

File: src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/POLineDetails.js [hunk 3]
```diff
@@ -35,12 +40,22 @@ export const POLineDetails = ({ mappingDetails }) => {
...
+  const checkinItemsValue = getFieldValue(mappingDetails, 'checkinItems', 'value');
+  const cancellationDescription = getFieldValue(mappingDetails, 'cancellationRestrictionNote', 'value');
   const poLineDescription = getFieldValue(mappingDetails, 'poLineDescription', 'value');
 
   const automaticExportCheckbox = renderCheckbox('order.poLineDetails.automaticExport', automaticExport);
 
+  const receivingWorkflow = useMemo(
+    () => {
+      if (checkinItemsValue === 'true') return formatMessage({ id: `${TRANSLATION_ID_PREFIX}.order.poLineDetails.field.independent` });
+      if (checkinItemsValue === 'false') return formatMessage({ id: `${TRANSLATION_ID_PREFIX}.order.poLineDetails.field.synchronized` });
```
Comment (artem-blazhko) [inline src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/POLineDetails.js:52]: Use `BOOLEAN_STRING_VALUES` constant

**Review by artem-blazhko — APPROVED** [review]

File: src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/EResourcesDetails.js (hunk 2 above)
Comment (mariia-aloshyna) [reply src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/EResourcesDetails.js]: Hi @artem-blazhko I added field maps for indexes, could you review, please?

File: src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/Location.js [hunk 4]
```diff
@@ -48,6 +57,32 @@ export const Location = ({
...
+    () => {
+      const onLocationAdd = (fieldsPath, refTable, fieldIndex, isFirstSubfield) => {
+        const repeatableFieldActionPath = getRepeatableFieldName(fieldIndex);
+
+        handleRepeatableFieldAndActionAdd(repeatableFieldActionPath, fieldsPath, refTable, setReferenceTables, isFirstSubfield);
+      };
+
+      return onAdd(locations, 'locations', LOCATION_FIELDS_MAP.LOCATIONS, initialFields, onLocationAdd, 'order');
+    },
+    [initialFields, locations, setReferenceTables],
```
Comment (zburke) [inline src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/Location.js]: Add `LOCATION_FIELDS_MAP.LOCATIONS` to the dependency array here and elsewhere where lint whines about new code.

**Review by zburke — CHANGES_REQUESTED** [review]
I have to be honest: there's so much going on in this PR, and I have so little context for it, that it's impossible for me to review effectively. I can say the introduction of maps to provide CONSTANT values for fields identified by numbers is a huge win. Thank you for that!

A few nits: a few test-related deps should be moved to dev-deps, and a few lint complaints about new code need to be addressed. (If lint already warns about something on master then you don't need to fix that here, but a PR should not add any new lint violations.) There is one typo to fix as well.

File: src/settings/MappingProfiles/detailsSections/constants.js [hunk 5]
```diff
@@ -99,3 +99,5 @@ export const VENDOR_VISIBLE_COLUMNS = {
   REF_NUMBER: 'refNumber',
   REF_NUMBER_TYPE: 'refNumberType',
 };
+
+export const QUOTED_ID_PATTERN = /^"([a-zA-Z0-9-]*)"$/;
```
[the remark this answers is not in the document]
Comment (mariia-aloshyna) [reply src/settings/MappingProfiles/detailsSections/constants.js:103]: This pattern is for validating a UUID wrapped in quotes, for example `'"6ec0bd7f-11c0-43da-975e-2a8ad9ebae0b"'`

**Comment by mariia-aloshyna** [comment]
Hi @zburke thank you for the review! All comments are fixed. I know there are a lot of changes, it's just improvements after implementing the main feature that's because it looks disorderly and massive, all of them are described in the story, I didn't want to duplicate them here

File: src/settings/MappingProfiles/detailsSections/constants.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (zburke) [reply src/settings/MappingProfiles/detailsSections/constants.js:103]: Exactly: that regex is way more permissive than "a UUID wrapped in quotes". Maybe at least change `a-zA-Z` to `a-fA-F`? Your call, but even that isn't very strict and tools that perform real validation are readily available.

**Review by zburke — APPROVED** [review]

File: src/settings/MappingProfiles/detailsSections/constants.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (mariia-aloshyna) [reply src/settings/MappingProfiles/detailsSections/constants.js:103]: Hi @zburke I changed the regex to a more strict

**Review by zburke — APPROVED** [review]
LGTM

**Pull request merged**